dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java

```diff
@@ -409,6 +422,19 @@ public static final class Delta {
             may = maySet;
             must = mustSet;
         }
+
+        private static Delta combine(List<Delta> deltas) {
+            if (deltas.size() == 1) {
+                return deltas.get(0);
+            }
+            EventGraph mayDelta = new EventGraph();
+            EventGraph mustDelta = new EventGraph();
```

Comment:
It will always create a new object, even if must sets are disabled and always point to the signleton. Probably, will not have any visible impact in practice.

Reply:
Yes, to streamline the handling of the `enableMustSets` option, I changed the code to just always compute the must sets and then just filter them out in the core algorithm before propagating.
This way the option is handled at a central place using only a few lines of code.
I believe the performance impact is marginal.